django/src/rdwatch/views/model_run.py

```diff
@@ -456,3 +518,29 @@ def get_modelrun_evaluations(request: HttpRequest, hyper_parameters_id: int):
         }
     query['region'] = model_run['region']
     return 200, query
+
+
+@router.get('/{id}/download')
+def download_annotations(request: HttpRequest, id: int):
+    # Needs to go through the siteEvaluations and download one for each file
+    site_evals = SiteEvaluation.objects.filter(configuration_id=id)
+    model_run_name = HyperParameters.objects.get(pk=id).title
+    with tempfile.TemporaryDirectory() as temp_dir:
+        zip_file_path = os.path.join(temp_dir, 'annotations.zip')
+        file_list = []
+        for item in site_evals:
+            data, site_id = get_site_model_feature_JSON(item.pk)
+            file_name = os.path.join(temp_dir, f'{site_id}.json')
+            with open(file_name, 'w') as file:
+                json.dump(data, file)
+            file_list.append({'filename': file_name, 'siteId': site_id})
+        with zipfile.ZipFile(zip_file_path, 'w') as zipf:
+            for item in file_list:
+                zipf.write(item['filename'], f"{item['siteId']}.json")
+
+        with open(zip_file_path, 'rb') as file:
+            response = HttpResponse(file.read(), content_type='application/zip')
+            response[
+                'Content-Disposition'
+            ] = f'attachment; filename="{model_run_name}.zip"'
+            return response
```

Comment:
For AE Ground Truth it's like 30 seconds (The largest region/model-run.  For most smaller model runs it's 2-5 seconds like shown in the demo video.  I wondered about moving this into a task where when complete it provides a download.

Reply:
Yeah I was going to suggest a task too. Someone could easily DDoS the server by attempting to download several model runs at once.

Reply:
I guess the plan would be:

1. endpoint which kicks of a task to generate a zip file for a model-run.  Returning the taskId for celery
2. Client polls the celery task-id to see if the task is finished.  Once finished it can ask for the file associated with the taskId
3. While running the task creates the zip file and stores it (not sure if I store it in S3 or on system)
4. The client once it knows the task is finished will request to download the file
5. After downloading the file it will remove it (either from system or S3).

Reply:
Looking into it again (I forgot the worker is in another container and could be in another machine without shared storage), the best/easiest option may be an 'AnnotationExports' table which utilizes S3 FileField to store the zip file.  It would contain the taskId, the model-run and a timestamp.  A third indicator will there to let it know if it's been downloaded and then we can use the schedule clean up task to remove downloaded files after an hour or so (whenever the scheduled clean up task runs).